pages/_app.js

```diff
@@ -107,6 +110,12 @@ export default function App({
               randomSurprise={randomSurprise}
               setRandomSurprise={setRandomSurprise}
               handleSurprise={handleSurprise}
+              isImage={isImage}
+              setIsImage={setIsImage}
+              publicId={publicId}
+              setPublicId={setPublicId}
+              isUploading={isUploading}
+              setIsUploading={setIsUploading}
```

Comment:
As mentioned before: Try to avoid  passing state setter functions as props. It's better to encapsulate them within handler functions.